flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -189,15 +180,17 @@ public boolean stopWorker(final KubernetesWorkerNode worker) {
 	public void onAdded(List<KubernetesPod> pods) {
 		runAsync(() -> {
 			for (KubernetesPod pod : pods) {
-				if (numPendingPodRequests > 0) {
-					numPendingPodRequests--;
+				WorkerResourceSpec workerResourceSpec = podWorkerResources.get(pod.getName());
```

Comment:
Can this be `null`?

Reply:
If we add a duplication check to filter out pods from previous attempt, then this should not be `null`. I'll add a `checkNotNull`.